### ISSUE TITLE
Curly quotes within words, such as contractions

### ISSUE THREAD

**Issue by jmakeig** [swyphcosmo/vscode-spellchecker#44]
`doesn't` is recognized as spelled correctly, however `doesn’t` is not. Note the curly apostrophe. Same with `"anything here"` and `“anything here”`. For the second case (i.e. `“anything here”`), I added `/[“”‘’]/g` to the list of regexps to ignore. That doesn’t seem to work for contractions, like doesn’t, though, where the punctuation is within the word itself.

- SpellChecker v1.2.13
- VSCode v1.13.1
- macOS 10.12.5 (16F73)

**Comment by swyphcosmo** [comment]
I apologize for not getting back to you sooner.

The plugin replaces any match from the regex with a space character. That's why it doesn't work with contractions. I have a special check for contractions. I will add the curly apostrophe character to this check, and release an update soon.

**Comment by swyphcosmo** [comment]
Actually, as I was looking into this, the curly apostrophe is already included in the current version of the plugin. The regex you added removes them before the extra checks catch them. If you change your regex to `/[“”]/g`, you should find that `doesn’t` is properly recognized.  Please let me know if that's not the case.

**Comment by jmakeig** [comment]
I’m still having some issues with the handling of non-ASCII punctuation. For example the plural possessive `others’` is still flagged as incorrectly spelled. Here’s my ignore configuration:

```javascript
"spellchecker.ignoreRegExp": [
  "/<a name=[\"']?.+[\"']?></a>/g", // HTML anchors
  "/\\]\\([^\\(]+\\)/g", // Markdown links
  "/\\*\\*TODO:?.*\\*\\*/g", // TODOs
  "/\\b[A-Z]+s?\\b/g", // Acronyms and their plurals
  "/\\b[A-Z](?:[A-Z0-9]*[a-z][a-z0-9]*[A-Z]|[a-z0-9]*[A-Z][A-Z0-9]*[a-z])[A-Za-z0-9]*\\b/g", // CamelCase
  "/[–—“”…]/g" // Special characters
],
```

**Comment by swyphcosmo** [comment]
I think this is just an issue with plural possessives using curly apostrophes. The current searches for non letter characters will keep apostrophes only in contractions, i.e. apostrophes with a letter before and after it. For `others'`, the apostrophe is replaced by a space with the internal regex. For `others’`, your regex replaces the curly apostrophe with an ASCII apostrophe. Since user regex's are processed after the internal ones, the ASCII apostrophe is never replaced by a space. 

This extension is based on the [Hunspell](http://hunspell.github.io/) spell checker, which only spell checks and does not perform grammar checks. It knows that possessives typically end with `'s`, but it doesn't know the grammar rules that if the character before the apostrophe is an `s`, then you remove the trailing `s`.

**Comment by jmakeig** [comment]
> For others’, your regex replaces the curly apostrophe with an ASCII apostrophe.

I don’t have a user ignore regex that matches single curly quotes, do I? (Am I misreading my config above?) Shouldn’t all leading or trailing punctuation just be ignored, thus eliminating the need for my `"/[–—“”…]/g"` ignore? (Same problem with ellipses and dash characters as trailing apostrophes.)
